### DISCUSSION TITLE
How does the file supervisor work?

### DISCUSSION THREAD

**Discussion by aleccool213** [ojkelly/yarn.build#100] [Q&A] [answered]
After reading the docs, I think I have an idea of how this package should work. Please correct any wrong assumptions.

What I'm thinking this package will achieve is reduced build times locally due to `yarn build` understanding what has been modified. What parameters does this package consider for a module to be modified? 

For example, if I have a workspace with the package `user`, I run `yarn build` and get 10 seconds (it built from scratch). But when I run it again with no code changed, I see it runs the entire build from scratch again (10 seconds). Is there a file supervisor seeing if there are any changes between the last two runs?

**Comment by ojkelly** [comment, marked as answer]
Currently it looks at the `main` script in `package.json` to determine the build folder.

There’s a ticket #3 is open to allow for more detailed customised options. I want to allow you to define the input and output folders for your build in `package.json`.

Additionally, the supervisor itself keeps a file `.yarn/local-build-cache.json`. This keeps track of the `lastModified` time of the package folder, excluding the build output.

**Comment by aleccool213** [comment]
@ojkelly 

After I run `yarn build`, no `.yarn/local-build-cache.json` gets created on my local. Is that a bug?

**Comment by ojkelly** [comment]
It was renamed in a recent commit https://github.com/ojkelly/yarn.build/blob/trunk/packages/plugins/plugin-build/src/commands/supervisor/index.ts#L25 to `.yarn/yarn.build.json` because it contains a bunch more info.
